### ISSUE TITLE
Potential Race Condition for MP-285

### ISSUE THREAD

**Issue by AdvancedImagingUTSW** [TheDeanLab/ASLM#970]
When interrupting stage movement, which is often used to update the position of the stage, there is a race condition between the model and the event triggered by the GUI action. Returned response does not make sense, and is a good candidate for disrupting operation of the stage itself. Low priority, works 99.99% of the time, user just needs to reset the device.

Manual here: https://www.sutter.com/manuals/MP-285A_RefMan.pdf

Page 64.

**Comment by AdvancedImagingUTSW** [comment]
Annie has had a chance to look into this. 

```
8.8.1 Interrupt Move (^C) Command
This command interrupts and stops a move in progress that originally initiated by the Move
(‘m’) command. The command sequence consists of 1 byte: Command byte (no terminator).
Return data consists of 1 byte if movement is not in progress, or 2 bytes (‘
= ’ (move-in-progress indicator) and task-complete indicator.
```

The interrupt move command does actually stop the movement of the stage. But rather, while the stage is in movement, and we ask for the current position, we get the response for the interrupt move first then the current position, and then the response that the movement has ended. But the sequence can change depending upon the timing of the call relative to the end movement.

**Comment by AdvancedImagingUTSW** [comment]
Fixed with #1006
